conan/tools/intel/inteloneapi.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+    Intel generator module for oneAPI Toolkits.
+
+    For simplicity and clarity, Intel informally refers to some of the terms in this document,
+    as listed below:
+
+        - ICX - Intel oneAPI DPC++/C++ Compiler
+        - ICC Classic - Intel C++ Compiler Classic
+        - DPCPP - Intel oneAPI DPC++/C++ Compiler
+
+    DPCPP is built upon ICX as the underlying C++ Compiler, therefore most of this information
+    also applies to DPCPP.
+
+    Intel oneAPI Toolkit (DPC++/C++ Compiler)
+        - Versioning: https://software.intel.com/content/www/us/en/develop/articles/oneapi-toolkit-version-to-compiler-version-mapping.html
+        - Compiler: https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top.html
+"""
+import os
+import platform
+import textwrap
+
+from conan.tools.env.environment import create_env_script
+from conans.client.tools.win import is_win64, _system_registry_key
+from conans.errors import ConanException
+
+
+def is_using_intel_oneapi(compiler_version):
+    """Check if the Intel compiler to be used belongs to Intel oneAPI
+
+    Note: Intel oneAPI Toolkit first version is 2021.1
+    """
+    return int(compiler_version.split(".")[0]) >= 2021
+
+
+def get_inteloneapi_installation_path():
+    system = platform.system()
+    # Let's try the default dirs
+    if system == "Windows":
+        if is_win64():
+            base = r"SOFTWARE\WOW6432Node"
+        else:
+            base = r"SOFTWARE"
+        path = r"{base}\Intel\Products\IntelOneAPI".format(base=base)
+        from six.moves import winreg  # @UnresolvedImport
```

Comment:
You might avoid ``six`` usage now, and fail without managing error if Py2, as this will directly fail when merging this branch to ``develop2`` otherwise.